IHP/View/Form.hs

```diff
@@ -759,6 +759,86 @@ selectField field items = FormField
+--
+-- You also need to define a instance @CanSelect User@:
+--
+-- > instance CanSelect User where
+-- >     -- Here we specify that the <option> value should contain a `Id User`
+-- >     type SelectValue User = Id User
+-- >     -- Here we specify how to transform the model into <option>-value
+-- >     selectValue user = user.id
+-- >     -- And here we specify the <option>-text
+-- >     selectLabel user = user.name
+--
+-- Given the above example, the rendered form will look like this (omitting classes for brevity):
+--
+-- > <!-- Assuming: users = [User { id = 1, name = "Marc" }, User { id = 2, name = "Andreas" }] -->
+-- > <form ...>
+-- >     <fieldset>
+-- >         <div>
+-- >           <input type="radio" id="option1" value="1"/>
+-- >           <label for="option1">Marc</label>
+-- >         </div>
+-- >         <div>
+-- >           <input type="radio" id="option2" value="2"/>
+-- >           <label for="option2">Andreas</label>
+-- >         </div>
+-- >     </fieldset>
+-- > </form>
+--
+-- If you want a certain value to be preselected, set the value in the controller. For example, to have the first user be preselected in the above example:
+--
+-- > action NewProjectAction = do
+-- >     users <- query @User |> fetch
+-- >     let userId = headMay users |> maybe def (.id)
+-- >     let target = newRecord @Project |> set #userId userId
+-- >     render NewView { .. }
+radioField :: forall fieldName model item.
+    ( ?formContext :: FormContext model
+    , HasField fieldName model (SelectValue item)
+    , HasField "meta" model MetaBag
+    , KnownSymbol fieldName
+    , KnownSymbol (GetModelName model)
+    , CanSelect item
+    , InputValue (SelectValue item)
+    ) => Proxy fieldName -> [item] -> FormField
+radioField field items = FormField
+        { fieldType =
+            let
+                itemToTuple :: item -> (Text, Text)
+                itemToTuple item = (selectLabel item, inputValue (selectValue item))
+            in
+                 RadioInput (map itemToTuple items)
+        , fieldName = cs fieldName
+        , fieldLabel = removeIdSuffix $ fieldNameToFieldLabel (cs fieldName)
+        , fieldValue = inputValue ((getField @fieldName model :: SelectValue item))
+        , fieldInputId = cs (lcfirst (getModelName @model) <> "_" <> cs fieldName)
+        , validatorResult = getValidationViolation field model
+        , fieldClass = ""
+        , labelClass = ""
+        , disabled = False
+        , disableLabel = False
+        , disableGroup = False
+        , disableValidationResult = False
+        , additionalAttributes = []
+        , cssFramework = ?formContext.cssFramework
+        , helpText = ""
+        , placeholder = ""
+        , required = False
+        , autofocus = False
+    }
+    where
+        fieldName = symbolVal field
+        FormContext { model } = ?formContext
+{-# INLINE radioField #-}
```

Comment:
What's the purpose of `INLINE`?

Reply:
When the code is inlined, this makes it easier for GHC to optimize the other very generic code. E.g. the code `cs (lcfirst (getModelName @model) <> "_" <> cs fieldName)` can be simplified a lot when the `model` type variable is statically known (which is typically the case when the helper is inlined at the right place)

Reply:
How do you know when to use it? Is there some rule of thumb?

Reply:
There's no hard rule. I've discovered some of those places by dumping the GHC Core output and checking for potential places. GHC can also report some of those places using the `missed-specialisations` flag (it's enabled in IHP)